flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobClient.java

```diff
@@ -95,7 +95,7 @@ public BlobClient(InetSocketAddress serverAddress, Configuration clientConfig)
             // the
             // InetSocketAddress can cache a failure in hostname resolution forever.
             socket.connect(
-                    new InetSocketAddress(serverAddress.getHostName(), serverAddress.getPort()),
+                    new InetSocketAddress(serverAddress.getHostString(), serverAddress.getPort()),
```

Comment:
The change looks good. nit: I wonder if we could add a comment to explain why we do. not use getHostName. Maybe include some of the following reasons:  Addresses are often configured as IP addresses for reliability
DNS may be unavailable or slow in containerized/cloud environments
Every blob transfer would trigger unnecessary reverse DNS lookups
Connection failures could occur if DNS resolution fails